packages/@sanity/base/src/components/formField/FormFieldSet.tsx

```diff
@@ -107,32 +106,35 @@ export const FormFieldSet = forwardRef(
     const hasValidations = validation.length > 0
     const forwardedRef = useForwardedRef(ref)
 
+    const handleToggleCollapse = useCallback(() => {
+      setCollapsed(!collapsed)
+      if (onToggle) onToggle(!collapsed)
+    }, [collapsed, onToggle])
+
+    let getContent = () => (
+      <Grid columns={columns} gapX={4} gapY={5}>
+        {getChildren(children)}
+      </Grid>
+    )
```

Comment:
I’m wondering if wrapping this is in `useMemo` might be a small optimization improvement here, and also increase readability somewhat. Something like:

```tsx
const content = useMemo(() => {
  if (collapsed) return null

  if (changeIndicator) {
    const changeIndicatorProps = typeof changeIndicator === 'object' ? changeIndicator : {}

    return <ChangeIndicator {...changeIndicatorProps}>{getChildren(children)}</ChangeIndicator>
  }

  return (
    <Grid columns={columns} gapX={4} gapY={5}>
      {getChildren(children)}
    </Grid>
  )
}, [changeIndicator, children, collapsed])
```

Reply:
Great point! Fixed in https://github.com/sanity-io/sanity/pull/2482/commits/4922c7279b4fb1c6003e376ac453278e9156a817